stan/math/rev/core/operator_addition.hpp

```diff
@@ -116,10 +138,153 @@ inline var operator+(const var& a, Arith b) {
  */
 template <typename Arith, require_arithmetic_t<Arith>* = nullptr>
 inline var operator+(Arith a, const var& b) {
-  if (a == 0.0) {
-    return b;
+  return b + a;
+}
+
+/**
+ * Addition operator for matrix variables (C++).
+ *
+ * @tparam VarMat1 A matrix of vars or a var with an underlying matrix type.
+ * @tparam VarMat2 A matrix of vars or a var with an underlying matrix type.
+ * @param a First variable operand.
+ * @param b Second variable operand.
+ * @return Variable result of adding two variables.
+ */
+template <typename VarMat1, typename VarMat2,
+          require_all_rev_matrix_t<VarMat1, VarMat2>* = nullptr>
+inline auto operator+(const VarMat1& a, const VarMat2& b) {
+  check_matching_dims("operator+", "a", a, "b", b);
+  using op_ret_type = decltype(a.val() + b.val());
```

Comment:
I think you either need to change this to:
```
plain_type_t<decltype(a.val() + b.val())>;
```

Or remove the ```.eval()``` from ```ret```. Otherwise ```op_ret_type``` will be a ```CWiseBinaryOp``` while the result of ```(a.val() + b.val()).eval()``` will be a plain object. Unless ```promote_var_matrix_t``` automatically returns a plain type?

Reply:
> Unless promote_var_matrix_t automatically returns a plain type?

Yep that's what happens!